### PR TITLE
update(RxCBCentralManager): improve BluetoothManager.listenOnRestoredState()

### PR DESCRIPTION
**Pull request by HAKASHUN**
Changed willRestoreStateSubject to  ReplaySubject ~~BehaviorSubject~~ from PublishSubject.

for issue https://github.com/Polidea/RxBluetoothKit/issues/101

### CODE REVIEW

File: Source/RxCBCentralManager.swift [hunk 1]
```diff
@@ -53,7 +53,7 @@ class RxCBCentralManager: RxCentralManagerType {
 
     @objc private class InternalDelegate: NSObject, CBCentralManagerDelegate {
         let didUpdateStateSubject = PublishSubject<BluetoothState>()
-        let willRestoreStateSubject = PublishSubject<[String: Any]>()
+        let willRestoreStateSubject = BehaviorSubject<[String: Any]?>(value: nil)
```
Comment (Kacper20) [inline Source/RxCBCentralManager.swift]: Couldn't this be a ReplaySubject instance with buffer size 1? It doesn't need the starting value.

Comment (HAKASHUN) [reply Source/RxCBCentralManager.swift]: Thanks for your review!
I'll try the change and check behavior on my project as soon as possible.

Comment (HAKASHUN) [reply Source/RxCBCentralManager.swift]: @Kacper20 I tried your suggestion and the changes worked well!

**Comment by Kacper20** [comment]
Looks good, thanks! Merging it :)

**Comment by HAKASHUN** [comment]
Thanks!

**Pull request merged**